### PR TITLE
Move non-go related files to the `_assets` folder. #621

### LINKED ISSUE #621: Move non-go related files to the `_assets` folder.
# Problem

Some of the directories in our root dir are valid go packages. Some aren't.
That makes it harder to understand what is what.

# Implementation

Let's create an `_assets` directory and group these files/packages there (as much as we can).
The proposal looks like that.
```
SOURCE                        TARGET
.gitignore
.gometalinter.json       * -> _assets/
.travis.yml              *
DEPENDENCIES.md          * -> _assets/docs/
Dockerfile               * -> _assets/build/
Gopkg.lock
Gopkg.toml
Jenkinsfile              * -> _assets/ci/
Jenkinsfile-manual       * -> _assets/ci/
LICENSE.md
Makefile
README.md
build                    * -> _assets/build/
ci                       * -> _assets/ci/
cmd
docker                   * -> _assets/compose/
extkeys
geth-patches             * -> _assets/patches/geth
lib
metrics
package-lock.json        * -> _assets/
package.json             * -> _assets/
profiling
static
vendor
xgo                      * -> _assets/build/
```

# Acceptance Criteria

**All** make commands are passing and providing the same results as before the changes (see `Makefile`).
Link in the .md files work.

### PR DESCRIPTION
**Pull request by canercidam**
Reorganizes structure as specified in the issue. 

- [x] No conflicts with #624.
- [x] All make targets work.

Closes #621

### CODE REVIEW

**Comment by canercidam** [comment]
Tested almost all make targets in a docker container using:
```Dockerfile
FROM golang:1.9

WORKDIR /go/src/github.com/status-im
RUN git clone https://github.com/canercidam/status-go -b issue-621 --depth 1
WORKDIR status-go
COPY . .
RUN chmod +x try

CMD ["./try"]
```

And I had to skip these:
```
make docker-image
make xgo-docker-images
make xgo
make test-e2e
```

**Comment by canercidam** [comment]
@mandrigin Ready for review.

File: _assets/patches/geth/0008-tx-pool-nonce.patch [hunk 1]
```diff
@@ -0,0 +1,19 @@
+diff --git a/internal/ethapi/api.go b/internal/ethapi/api.go
```
Comment (mandrigin) [inline _assets/patches/geth/0008-tx-pool-nonce.patch]: hmm, was this file added?

**Review by mandrigin — CHANGES_REQUESTED** [review]
One question :)

File: _assets/patches/geth/0008-tx-pool-nonce.patch (hunk 1 above)
Comment (canercidam) [reply _assets/patches/geth/0008-tx-pool-nonce.patch]: That was a conflict after rebase and I couldn't make sense of it. Was it removed the last time?

Comment (canercidam) [reply _assets/patches/geth/0008-tx-pool-nonce.patch]: It doesn't seem to exist on `develop`. I'll just remove it then.

Comment (mandrigin) [reply _assets/patches/geth/0008-tx-pool-nonce.patch]: Yes, I don't see it on the tip of `develop` anymore.

**Review by mandrigin — APPROVED** [review]

File: Makefile [hunk 2]
```diff
@@ -18,6 +18,7 @@ XGOIMAGE = statusteam/xgo:$(XGOVERSION)
 XGOIMAGEIOSSIM = statusteam/xgo-ios-simulator:$(XGOVERSION)
 
 DOCKER_IMAGE_NAME ?= status-go
+DOCKER_FILE_PATH ?= --file _assets/build/Dockerfile
```
Comment (adambabik) [inline Makefile]: Has it been tested that building a docker image (`make docker-image`) still works as expected?

File: Makefile [hunk 3]
```diff
@@ -72,32 +73,32 @@ statusgo-library: ##@cross-compile Build status-go as static library for current
...
 docker-image: ##@docker Build docker image (use DOCKER_IMAGE_NAME to set the image name)
 	@echo "Building docker image..."
-	docker build --build-arg "build_tags=$(BUILD_TAGS)" . -t $(DOCKER_IMAGE_NAME)
+	docker build --file $(DOCKER_FILE_PATH) --build-arg "build_tags=$(BUILD_TAGS)" . -t $(DOCKER_IMAGE_NAME)
 
 xgo-docker-images: ##@docker Build xgo docker images
 	@echo "Building xgo docker images..."
-	docker build xgo/base -t $(XGOIMAGE)
-	docker build xgo/ios-simulator -t $(XGOIMAGEIOSSIM)
+	docker build --file $(DOCKER_FILE_PATH) _assets/build/xgo/base -t $(XGOIMAGE)
```
Comment (adambabik) [inline Makefile]: `--file` here is not needed. It should build a docker image based on `_assets/build/xgo/base/Dockerfile`.

File: Makefile [hunk 4]
```diff
@@ -72,32 +73,32 @@ statusgo-library: ##@cross-compile Build status-go as static library for current
...
+	docker build --file $(DOCKER_FILE_PATH) _assets/build/xgo/ios-simulator -t $(XGOIMAGEIOSSIM)
```
Comment (adambabik) [inline Makefile]: The same here but `_assets/build/xgo/ios-simulator/Dockerfile`.

**Review by adambabik — CHANGES_REQUESTED** [review]
Ideally, each target changed in `Makefile` should be run and verified. For `xgo`, try only `xgo/base`.

**Comment by canercidam** [comment]
Building `xgo/base` and `make docker-image` are working. Thanks @adambabik, I missed the point there for a moment. I'm going to do a last run on all targets before merge.

One last thing though: Can I suggest leaving `.gometalinter.json` at the root? I find it especially useful while working on the project (in subpackages), since triggering gometalinter from VSCode naturally uses `.gometalinter.json` at the root. @mandrigin @adambabik

**Comment by mandrigin** [comment]
@canercidam sure, no problem if it helps some tooling to work out of the box!

**Comment by canercidam** [comment]
Alright, I'm positive enough that all make targets are working. I also tried `statusgo-android` which depends on `xgo` target.

**Review by adambabik — APPROVED** [review]
Thanks! It's so clean now! :D

**Comment by mandrigin** [comment]
@canercidam thanks for improving our repo! ⭐️

**Pull request merged**